src/borg/testsuite/archiver/create_cmd.py

```diff
@@ -731,6 +771,44 @@ def test_common_options(self):
         log = self.cmd(f"--repo={self.repository_location}", "--debug", "create", "test", "input")
         assert "security: read previous location" in log
 
+    def test_hashing_time(self):
+        def extract_hashing_time(borg_create_output):
+            borg_create_output = borg_create_output.strip().splitlines()
+            borg_create_output = [line.split(":", 1) for line in borg_create_output]
+            hashing_time = [line for line in borg_create_output if line[0] == "Time spent in hashing"].pop()
+            hashing_time = hashing_time[1]
+            hashing_time = float(hashing_time.rstrip(" seconds"))
+            return hashing_time
+
+        # Test case set up: create a repository and a 500 files
+        self.cmd(f"--repo={self.repository_location}", "rcreate", RK_ENCRYPTION)
+        for i in range(0, 500):
+            self.create_regular_file(f"testfile{i}")
```

Comment:
how long will that file be?

Reply:
Zero bytes long. Still, I'm not sure what to do here. In my computer if a create a single file like this: `self.create_regular_file("testfile", size=50000000)` it fails the test.

Reply:
if the file is 0 bytes long, there is nothing to hash, so you can't expect the hashing time to be more than 0.0 seconds.

find out, why the test fails. if an assertion fails, you can usually see the output of any `print()` you did before.

after debugging the root cause, please use less than 50MB for the test. the tests shall run rather quickly, so do not produce/use more data than needed.

Reply:
I just noticed that the function `id_hash` method is given a file path as its `data` argument, not the contents of the file.
That is why it fails with a single file 50MB long and it passes with 1000 files

Reply:
id_hash is used for multiple purposes. for the main purpose, it gets file content chunks.

but how about just not giving it an empty file?

Reply:
btw, due to deduplication, avoid giving it a lot of duplicate data (or it will do less than you might think).

you can avoid dedup by giving it random data.

Reply:
Ok, I understand where I'm wrong now. Up to this point, the only hashing I measure in my code is in `archive.py` line 1441. This is what I mentioned in my previous comment about only file path being hashed.
So, I am missing the measurement of the hashing time done to process file chunks right? This is done in the method `ChunksProcessor.process_fille_chunks`

Reply:
yes. for that, the file does need content (> 0 bytes length). Also, if you must produce more than a single chunk, avoid using all-zero bytes, but use random data to make borg really compute hashes (and not do some clever optimisation).